### ISSUE TITLE
Config file is not loaded

### ISSUE THREAD

**Issue by mpwp** [yi-editor/yi#965]
I'm trying to update yi with cabal and it seems that the config file is not loaded, although every packages are installed without errors. I cleaned the local package database and made a new install (of the yi suite) but it isn't working.
I just put the *yi-vim-vty-dynamic* example config, nothing custom.

I'm on voidlinux with ghc 8.0.1 and cabal 1.24.0.2.
I don't plan to install stack just for yi but it could be possible.

`yi -v` gives me version 0.13.0.1 while yi-core-0.13.5 is installed.
Did I miss something? If yes, what? Otherwise how can I correct that?

As a side note, I worked as a technical writer for one year I'm considering to make some documentation for yi, how can I help? Where can I start?

Thanks a lot!

**Comment by noughtmare** [comment]
__TL;DR:__ I recommend using the static yi versions to avoid all this.

I will explain how the compilation of yi works and how the dynamic configurations tie into this. 

Yi is basically a library which you can use to build your own editor. The configuration file is a perfectly valid cabal project which must be built like any other package. In this package you can specify how you want your Yi to look like. You can even completely change the editor to the point that it would not be recognizable as Yi.

For this to work you need to recompile Yi for every change you make. The static configurations make this obvious by forcing you to manually recompile your package for every change you make. The dynamic configurations use [dyre](https://hackage.haskell.org/package/dyre) to try to automate this process.

The dynamic configurations basically load the package that is at the location `~/.config/yi` and compile that and transfer the state of the current Yi to the newly compiled Yi (you can do this in the editor with the `reload` command). This is also automatically done when you start Yi and it detects that the contents of the `~/.config/yi` have been changed.

__I think your problem is in the following step:__

To achieve dynamic recompilation the you have to kickstart the process by compiling a dynamic configuration manually first. The default yi package is a static configuration so it won't check for changes in the `.config/yi` folder.

__My reasoning behind the switch from dynamic to static (this is a recent development):__

The dyre package that is used to provide this dynamic recompilation is derived from the way that xmonad handles recompilation. Xmonad is a window manager so it's not restarted very often and it is pretty inconvenient to restart so I think dynamic recompilation is very useful in that case. But Yi is a text editor and I personnally restart my text editor very often so I think that Yi should not use a dynamic recompilations system unless it is very intuitive and stable. 

__About the help with documentation:__

You're very welcome to help us with writing documentation. I think the github readme is outdated. I think the most extensive source of information is (or should become) the github page: [yi-editor.github.io](https://yi-editor.github.io). The github page contain some useful information, but I think there's still a lot missing (like the information in this comment). If you have any questions about documentation I think we should not accumulate them on this issue tracker but rather on [the issue tracker of yi-editor.github.io](https://github.com/yi-editor/yi-editor.github.com/issues) (except for issues relating the the README.md file here). I will gladly help you with any problems and questions you have about the Yi editor. If you have a quick question you can also message the IRC (freenode #yi) which I am usually monitoring (although it might take some time for me to answer there).

**Comment by mpwp** [comment]
Awesome!

I managed to make it worked with a static configuration, and just added Latex Mode in less time that it takes to compile.
The switch from dynamic to static is perfectly coherent (I use xmonad too).

Thanks a lot!
